component/mcsd/component_test.go

```diff
@@ -3,66 +3,118 @@ package mcsd
+	rootDirHistoryResponseBytes, err := os.ReadFile("test/root_dir_history_response.json")
 	require.NoError(t, err)
+	rootDirHistoryResponse := string(rootDirHistoryResponseBytes)
 
 	rootDirMux := http.NewServeMux()
 	rootDirMux.HandleFunc("/_history", func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "application/fhir+json")
 		w.WriteHeader(http.StatusOK)
-		_, _ = w.Write(testDataJSON)
+		_, _ = w.Write([]byte(rootDirHistoryResponse))
 	})
 	rootDirServer := httptest.NewServer(rootDirMux)
 
+	org1DirHistoryResponseBytes, err := os.ReadFile("test/org1_dir_history_response.json")
+	require.NoError(t, err)
+	org1DirHistoryResponse := string(org1DirHistoryResponseBytes)
+
+	org1DirMux := http.NewServeMux()
+	org1DirMux.HandleFunc("/fhir/_history", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/fhir+json")
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write([]byte(org1DirHistoryResponse))
+	})
+	org1DirServer := httptest.NewServer(org1DirMux)
+
+	orgDir1BaseURL := org1DirServer.URL + "/fhir"
+	rootDirHistoryResponse = strings.ReplaceAll(rootDirHistoryResponse, "{{ORG1_DIR_BASEURL}}", orgDir1BaseURL)
+	org1DirHistoryResponse = strings.ReplaceAll(org1DirHistoryResponse, "{{ORG1_DIR_BASEURL}}", orgDir1BaseURL)
+
 	localClient := &test.StubFHIRClient{}
-	component := Component{
-		config: Config{
-			RootDirectories: map[string]DirectoryConfig{
-				"rootDir": {
-					FHIRBaseURL: rootDirServer.URL,
-				},
-			},
-			LocalDirectory: DirectoryConfig{
-				FHIRBaseURL: "http://example.com/local/fhir",
+	component := New(Config{
+		RootAdminDirectories: map[string]DirectoryConfig{
+			"rootDir": {
+				FHIRBaseURL: rootDirServer.URL,
 			},
 		},
-		fhirClientFn: func(baseURL *url.URL) fhirclient.Client {
-			if baseURL.String() == rootDirServer.URL {
-				return fhirclient.New(baseURL, http.DefaultClient, nil)
-			}
-			if baseURL.String() == "http://example.com/local/fhir" {
-				return localClient
-			}
-			panic("unknown base URL: " + baseURL.String())
+		QueryDirectory: DirectoryConfig{
+			FHIRBaseURL: "http://example.com/local/fhir",
 		},
+	})
+	unknownFHIRServerClient := &test.StubFHIRClient{
+		Error: errors.New("404 Not Found"),
+	}
+	component.fhirClientFn = func(baseURL *url.URL) fhirclient.Client {
+		if baseURL.String() == rootDirServer.URL ||
+			baseURL.String() == orgDir1BaseURL {
+			return fhirclient.New(baseURL, http.DefaultClient, nil)
+		}
+		if baseURL.String() == "http://example.com/local/fhir" {
+			return localClient
+		}
+		t.Log("Using unknown FHIR server client for baseURL: " + baseURL.String())
+		return unknownFHIRServerClient
 	}
 	ctx := context.Background()
 
 	report, err := component.update(ctx)
 
 	require.NoError(t, err)
 	require.NotNil(t, report)
-	require.NoError(t, report[rootDirServer.URL].Error)
-	require.Empty(t, report[rootDirServer.URL].Warnings)
+	t.Run("assert sync report from root directory", func(t *testing.T) {
```

Comment:
It would be nice if the test also proves that errors and warnings work as intended in the report